assignments/002-unix-intro/unix-intro.sh

```diff
@@ -0,0 +1,550 @@
+
+
+#### Ex. 35: Count the number of lines/words/characters (using `wc`)
+####         in the two SAM files in the `sam` directory
+
+(base) [saw97@colostate.edu@login11 002-unix-intro]$ wc sam/DPCh_plate1_A05_S5.sam
+ 14219  45092 511203 sam/DPCh_plate1_A05_S5.sam
+(base) [saw97@colostate.edu@login11 002-unix-intro]$ wc sam/DPCh_plate1_A06_S6.sam
+ 14150  43985 479977 sam/DPCh_plate1_A06_S6.sam
+
+#### Ex. 36: Note that gzcat will decompress all files you give itand send the 
+####         output to _stdout_. Knowing this, count how many lines are in each of 
+####         the `R1` files in `fastq` (after decompressing them into text files),
+####         and also count up all the lines in the `R1` files in `fastq`. 
+####         Note they should be the same (use zcat)
+
+(base) [saw97@colostate.edu@login11 002-unix-intro]$ zcat fastq/*R1* | wc
+  19364   19364 1678394
+
+#### Ex. 37: Now, gzcat all the `R1` files in `fastq` and redirect that
+####         to a file called `R1_all_via_gzcat.fq` in the top directory of the repo
+
+(base) [saw97@colostate.edu@login11 002-unix-intro]$ zcat fastq/*R1.fq.gz > ../../R1_all_via_gzcat.fq
+
+
+#### Ex. 38: People are usually familiar with using `cat` to catenate text
+####         files. It turns out that you can also catenate `gzip` compressed files. 
+####         The result is another properly gzipped file that can be decompressed. 
+####         Catenate all the gzipped `R1` files in `fastq` into a single (still gzipped)
+####         file called `R1_all_via_cat.fq.gz`
+
+(base) [saw97@colostate.edu@login11 con-gen-csu]$ cat assignments/002-unix-intro/fastq/*R1.fq.gz >R1_all_via_cat.fq.gz
+(base) [saw97@colostate.edu@login11 con-gen-csu]$ ls
+R1_all_via_cat.fq.gz  R1_all_via_gzcat.fq  README.md  assignments  con-gen-csu.Rproj  data  dev
+
+
+#### Ex. 39: Copy `R1_all_via_cat.fq.gz` to `copy_of_R1_all_via_cat.fq.gz`
+
+(base) [saw97@colostate.edu@login11 con-gen-csu]$ cp R1_all_via_cat.fq.gz copy_of_R1_all_via_cat.fq.gz
+(base) [saw97@colostate.edu@login11 con-gen-csu]$ ls
+R1_all_via_cat.fq.gz  README.md    con-gen-csu.Rproj             data
+R1_all_via_gzcat.fq   assignments  copy_of_R1_all_via_cat.fq.gz  dev
+
+
+#### Ex. 40: Now, decompress `R1_all_via_cat.fq.gz` into `R1_all_via_cat.fq`
+
+(base) [saw97@colostate.edu@login11 con-gen-csu]$ zcat R1_all_via_cat.fq.gz > R1_all_via_cat.fq
+(base) [saw97@colostate.edu@login11 con-gen-csu]$ ls
+R1_all_via_cat.fq     R1_all_via_gzcat.fq  assignments        copy_of_R1_all_via_cat.fq.gz  dev
+R1_all_via_cat.fq.gz  README.md            con-gen-csu.Rproj  data
+
+
+#### Ex. 41: Compute the SHA1 hashes of `R1_all_via_cat.fq` and `R1_all_via_gzcat.fq`
+####         to confirm they are identical
+
+
+(base) [saw97@colostate.edu@login11 con-gen-csu]$ ls -R R1_all_via_cat.fq > cat_list.txt
+(base) [saw97@colostate.edu@login11 con-gen-csu]$ wc cat_list.txt
+ 1  1 18 cat_list.txt
+(base) [saw97@colostate.edu@login11 con-gen-csu]$ sha1sum cat_list.txt
```

Comment:
This was not quite what I had in mind, but no worries, because I got my head tied in knots around here when I was doing the homework, myself!